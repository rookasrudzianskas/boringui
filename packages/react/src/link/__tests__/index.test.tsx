import React from "react";
import {mount} from "enzyme";

import Link from "../index";

describe("Link", () => {
  it("should render correctly", () => {
    const wrapper = mount(
      <div>
        <Link href="https://nextui.org">link</Link>
        <Link color="secondary" href="https://nextui.org">
          link
        </Link>
        <Link isExternal color="success" href="https://nextui.org">
          link
        </Link>
        <Link underline href="https://nextui.org">
          link
        </Link>
        <Link block href="https://nextui.org">
          link
        </Link>
      </div>,
    );

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should be no errors when href missing", () => {
    const errorSpy = jest.spyOn(console, "error");
    const wrapper = mount(<Link />);

    expect(errorSpy).not.toHaveBeenCalled();
    expect(() => wrapper.unmount()).not.toThrow();
    errorSpy.mockRestore();
  });

  it("should forward ref", () => {
    let ref = React.createRef<HTMLAnchorElement>();
    const errorSpy = jest.spyOn(console, "error");
    const wrapper = mount(<Link ref={ref} />);

    expect(errorSpy).not.toHaveBeenCalled();
    expect(ref.current).not.toBeNull();
    expect(() => wrapper.unmount()).not.toThrow();
    errorSpy.mockRestore();
  });

  it("an warning should be thrown when using the pure prop", () => {
    const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {});

    mount(<Link pure />);
    expect(errorSpy).toHaveBeenCalled();
    errorSpy.mockRestore();
  });
});
