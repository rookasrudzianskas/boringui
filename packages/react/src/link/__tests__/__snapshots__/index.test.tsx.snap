// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `"<div><a class=\\"nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link\\" href=\\"https://nextui.org\\">link</a><a class=\\"nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-iVlgeG-color-secondary nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-iclraoi-css nextui-link\\" href=\\"https://nextui.org\\">link</a><a class=\\"nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-geYkrv-color-success nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-iighAhn-css nextui-link\\" href=\\"https://nextui.org\\">link<svg class=\\"nextui-c-cBXzrI nextui-link-icon\\" fill=\\"none\\" height=\\"1em\\" shape-rendering=\\"geometricPrecision\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" viewBox=\\"0 0 24 24\\" width=\\"1em\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg></a><a class=\\"nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-cSuHME-underline-true nextui-c-dIWzTP-isWIbj-animated-true nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link\\" href=\\"https://nextui.org\\">link</a><a class=\\"nextui-c-dIWzTP nextui-c-PJLV nextui-c-dIWzTP-hfnKRU-color-default nextui-c-dIWzTP-fZgbCn-block-true nextui-c-dIWzTP-isWIbj-animated-true nextui-c-dIWzTP-jSBLRj-cv nextui-c-PJLV-gulvcB-isFocusVisible-false nextui-c-dIWzTP-icTVmtU-css nextui-link\\" href=\\"https://nextui.org\\">link</a></div>"`;
